gapis/api/vulkan/frame_loop.go

```diff
@@ -709,3 +824,64 @@ func (f *frameLoop) resetDescriptorSetLayouts(ctx context.Context, stateBuilder
 
 	return nil
 }
+
+func (f *frameLoop) resetDescriptorPools(ctx context.Context, stateBuilder *stateBuilder) error {
+
+	// For every DescriptorSetLayout that was created during the loop...
+	for created := range f.descriptorPoolCreated {
+		// Write the command to delete the created object
+		descPool := GetState(f.loopEndState).descriptorPools.Get(created)
+		stateBuilder.write(stateBuilder.cb.VkDestroyDescriptorPool(descPool.Device(), descPool.VulkanHandle(), memory.Nullptr))
+	}
+
+	// For every DescriptorSetLayout that was destroyed during the loop...
+	for destroyed := range f.descriptorPoolDestroyed {
+		// Write the commands needed to recreate the destroyed object
+		stateBuilder.createDescriptorPoolAndAllocateDescriptorSets(GetState(f.loopStartState).DescriptorPools().Get(destroyed))
+
+		// Iterate through all the descriptor sets that we just recreated, adding them to the list of descriptor sets
+		// that need to be redefined.
+		descriptorPoolData := GetState(f.loopStartState).DescriptorPools().All()[destroyed]
+		for _, descriptorSetDataValue := range descriptorPoolData.DescriptorSets().All() {
+			containedDescriptorSet := descriptorSetDataValue.VulkanHandle()
+			f.descriptorSetChanged[containedDescriptorSet] = true
+		}
+	}
+
+	return nil
+}
+
+func (f *frameLoop) resetDescriptorSets(ctx context.Context, stateBuilder *stateBuilder) error {
+
+	// For every DescriptorSetLayout that was created during the loop...
+	for created := range f.descriptorSetCreated {
+		// Write the command to delete the created object
+		descSetObj := GetState(f.loopEndState).descriptorSets.Get(created)
+		handle := []VkDescriptorSet{descSetObj.VulkanHandle()}
+		stateBuilder.write(stateBuilder.cb.VkFreeDescriptorSets(descSetObj.Device(),
+			descSetObj.DescriptorPool(),
+			1,
+			stateBuilder.MustAllocReadData(handle).Ptr(),
+			VkResult_VK_SUCCESS))
+	}
+
+	// For every DescriptorSetLayout that was destroyed during the loop...
+	for destroyed := range f.descriptorSetDestroyed {
+		// Write the commands needed to recreate the destroyed object
+		descSetObj := GetState(f.loopStartState).descriptorSets.Get(destroyed)
+		descPoolObj := GetState(f.loopStartState).descriptorPools.Get(descSetObj.DescriptorPool())
+
+		descSetHandles := []VkDescriptorSet{descSetObj.VulkanHandle()}
+		descSetLayoutHandles := []VkDescriptorSetLayout{descSetObj.Layout().VulkanHandle()}
+		stateBuilder.allocateDescriptorSets(descPoolObj, descSetHandles, descSetLayoutHandles)
```

Comment:
Do we need to call `writeDescriptorSet` on the recreated descriptor set as well?